cmd/collector/app/builder/builder_flags.go

```diff
@@ -71,11 +71,11 @@ type CollectorOptions struct {
 func AddFlags(flags *flag.FlagSet) {
 	flags.Int(collectorQueueSize, app.DefaultQueueSize, "The queue size of the collector")
 	flags.Int(collectorNumWorkers, app.DefaultNumWorkers, "The number of workers pulling items from the queue")
-	flags.Int(collectorPort, ports.CollectorTChannel, "The TChannel port for the collector service")
-	flags.Int(collectorHTTPPort, ports.CollectorHTTP, "The HTTP port for the collector service")
-	flags.Int(collectorGRPCPort, ports.CollectorGRPC, "The gRPC port for the collector service")
-	flags.Int(collectorZipkinHTTPort, 0, "The HTTP port for the Zipkin collector service e.g. 9411")
-	flags.Bool(collectorGRPCTLS, false, "Enable TLS for the gRPC collector port")
+	flags.String(collectorTChanAddr, ports.GetAddressFromPort(ports.CollectorTChannel), "The TChannel address for the collector service")
+	flags.String(collectorHTTPAddr, ports.GetAddressFromPort(ports.CollectorHTTP), "The HTTP address for the collector service")
+	flags.String(collectorGRPCAddr, ports.GetAddressFromPort(ports.CollectorGRPC), "The gRPC address for the collector service")
+	flags.String(collectorZipkinHTTPAddr, ports.GetAddressFromPort(0), "The HTTP address for the Zipkin collector service")
```

Comment:
help strings should provide examples, e.g.

`The host:port (e.g. 127.0.0.1:5555 or :5555) of the collector's (HTTP|TChannel|gRPC|Zipkin) server`